### PR TITLE
[Central Beds] Address Jadu integration customer feedback.

### PR DESCRIPTION
**Pull request by neprune**
Closes https://github.com/mysociety/societyworks/issues/3792
Closes https://github.com/mysociety/societyworks/issues/3800
Closes https://github.com/mysociety/societyworks/issues/3801
Closes https://github.com/mysociety/societyworks/issues/3802

### CODE REVIEW

File: web/cobrands/fixmystreet-uk-councils/council_category_hide_element_rules.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+hide_element_rules = {
+    'Central Bedfordshire Council': {
+        'Fly Tipping': [
+            "description-tips",
```
Comment (davea) [inline web/cobrands/fixmystreet-uk-councils/council_category_hide_element_rules.js:4]: If this was a selector that could be passed straight to `$` on L692 of `web/cobrands/fixmystreet/fixmystreet.js` it'd be more flexible than only allowing element IDs.

File: templates/web/base/report/new/after_photo.html [hunk 2]
```diff
@@ -4,6 +4,6 @@
         <li>[% loc('For best results include a close-up and a wide shot') %]</li>
     </ul>
     <ul class="dont">
-        <li>[% loc('Avoid personal information and vehicle number plates') %]</li>
+        <li id="photo-tips-avoid-personal-info">[% loc('Avoid personal information and vehicle number plates') %]</li>
```
Comment (davea) [inline templates/web/base/report/new/after_photo.html:7]: There's a convention of using a `js-` prefix on IDs/classes that are only used for JS functionality, think that would be good to use here (and the other ones too).

File: web/cobrands/centralbedfordshire/flytipping.js [hunk 3]
```diff
@@ -52,4 +55,20 @@ $(fixmystreet).on('report_new:category_change', function() {
 
     checkAndToggle();
     witnessed.on('change', function() { checkAndToggle(); });
+}
+
+$(fixmystreet).on('report_new:category_change', function() {
+    if (!fixmystreet.bodies.includes('Central Bedfordshire Council')) {
```
Comment (davea) [inline web/cobrands/centralbedfordshire/flytipping.js:61]: `.includes` [isn't supported](https://caniuse.com/array-includes) in IE11, sadly, so think `.indexOf` is the way to go here.

```suggestion
    if (fixmystreet.bodies.indexOf('Central Bedfordshire Council') === -1) {
```

File: web/cobrands/fixmystreet/fixmystreet.js [hunk 4]
```diff
@@ -676,6 +678,30 @@ $.extend(fixmystreet.set_up, {
...
+        });
+        fixmystreet.hidden_elements = [];
+
+        // apply hide element rules
+        $.each(fixmystreet.bodies, function(index, body) {
+            if ( typeof hide_element_rules !== 'undefined' && hide_element_rules[body] && hide_element_rules[body][category] ) {
+                var elementIDs = hide_element_rules[body][category];
+                $.each(elementIDs, function(index, elementID) {
+                    var element = $('#' + elementID);
+                    if (!element) {
```
Comment (davea) [inline web/cobrands/fixmystreet/fixmystreet.js:693]: `element` is a jQuery object which is truthy, `.length` is the way to check if any elements matched:

```suggestion
                    if (!element.length) {
```

File: perllib/FixMyStreet/Cobrand/CentralBedfordshire.pm [hunk 5]
```diff
@@ -59,6 +59,30 @@ sub new_report_title_field_hint {
 
 sub report_sent_confirmation_email { 'external_id' }
 
+=item * We don't send report sent emails for Jadu backed contacts.
+
+=cut
+
+sub suppress_report_sent_email {
```
Comment (davea) [inline perllib/FixMyStreet/Cobrand/CentralBedfordshire.pm:66]: Is this new method needed? Can't you do the same check in the existing `report_sent_confirmation_email` method and return the appropriate value there?

**Review by davea — CHANGES_REQUESTED** [review]
A few little tweaks on the JS side of things, but also would be good to have some tests for the `suppress_reporter_alerts` functionality and `suppress_report_sent_email` (if the latter is staying).

File: perllib/FixMyStreet/Cobrand/CentralBedfordshire.pm (hunk 5 above)
Comment (neprune) [reply perllib/FixMyStreet/Cobrand/CentralBedfordshire.pm:66]: If `report_sent_confirmation_email` returns a falsey value, a report logged email is sent straight after a report is confirmed (`New.pm`). If truthy, a report logged email is sent after the report is sent (`Queue/Item/Report.pm`) instead, using the given string to extract an ID to include in the email.

Beds want no report logged email sent at all, hence the new method.

Comment (davea) [reply perllib/FixMyStreet/Cobrand/CentralBedfordshire.pm:66]: Ah, shame we can't have a trinary data type :) Makes sense! 👍

**Review by davea — APPROVED** [review]
Woo! Looks good 🎉

**Pull request merged**